garden_ai/pipelines.py

```diff
@@ -207,9 +229,26 @@ def garden_pipeline(
     def decorate(func):
         if model_connectors:
             metadata.models += [connector.metadata for connector in model_connectors]
+        metadata._as_step = Step(
+            function_text=inspect.getsource(func),
+            function_name=func.__name__,
+            description="Top level pipeline function",
+        )
+        metadata._target_garden_doi = garden_doi
         # let func carry its own metadata
-        func._pipeline_meta = metadata.dict()
-        func._garden_doi = garden_doi
+        func._garden_pipeline = metadata
+        return func
+
+    return decorate
+
+
+def garden_step(description: str = None):
+    def decorate(func):
+        func._garden_step = Step(
+            function_text=inspect.getsource(func),
+            function_name=func.__name__,
+            description=description,
```

Comment:
```suggestion
            description=description or func.__doc__,
```
might as well use a docstring for description if func has one

Reply:
maybe take or leave it, since we would already have a docstring show up in `function_text`, too

Reply:
I'm very open to this. Gonna leave it as is for now but we'll probably have another pass at this once we see it end to end in the UI